include/boost/uuid/detail/random_provider_detect_platform.hpp

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) 2017 James E. King III
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//   http://www.boost.org/LICENCE_1_0.txt)
+//
+// Platform-specific random entropy provider platform detection
+//
+
+#ifndef BOOST_UUID_DETAIL_RANDOM_PROVIDER_PLATFORM_DETECTION_HPP
+#define BOOST_UUID_DETAIL_RANDOM_PROVIDER_PLATFORM_DETECTION_HPP
+
+#include <boost/predef/library/c/gnu.h>
+#include <boost/predef/platform/cloudabi.h>
+// #include <boost/predef/os/bsd/open.h> re-enable when Boost.Predef #66 clears
```

Comment:
No need to wait for Predef. The functions we need are available on OpenBSD 2.5 and above.

Reply:
Good to know, I was planning on getentropy() there so... (on to the next comment)